This PR makes error messages easier to understand.

before:
```
panic: duplicate value 'esp32c3' in field : BuildTags

goroutine 1 [running]:
github.com/tinygo-org/tinygo/compileopts.(*TargetSpec).overrideProperties(0xc000353600, 0xc000353340)
        C:/tinygo/tinygo/compileopts/target.go:98 +0x848
github.com/tinygo-org/tinygo/compileopts.(*TargetSpec).resolveInherits(0xc000353340)
        C:/tinygo/tinygo/compileopts/target.go:157 +0x4d
github.com/tinygo-org/tinygo/compileopts.LoadTarget(0xc0000c9df0)
        C:/tinygo/tinygo/compileopts/target.go:223 +0x2b3
main.main()
        C:/tinygo/tinygo/main.go:1607 +0x3b70
```

after:
```
could not list target: C:\tinygo\tinygo\targets\esp32c3-12f.json : duplicate value 'esp32c3' in field : BuildTags
```

--- a/Makefile
+++ b/Makefile
@@ -419,6 +419,7 @@ tinygo-baremetal:
 .PHONY: smoketest
 smoketest:
 	$(TINYGO) version
+	$(TINYGO) targets > /dev/null
 	# regression test for #2892
 	cd tests/testing/recurse && ($(TINYGO) test ./... > recurse.log && cat recurse.log && test $$(wc -l < recurse.log) = 2 && rm recurse.log)
 	# compile-only platform-independent examples

--- a/compileopts/target.go
+++ b/compileopts/target.go
@@ -100,7 +100,7 @@ func (spec *TargetSpec) overrideProperties(child *TargetSpec) error {
 				}
 			}
 		default:
-			return fmt.Errorf("unknown field type : " + kind.String())
+			return fmt.Errorf("unknown field type: %s", kind)
 		}
 	}
 	return nil

--- a/compileopts/target.go
+++ b/compileopts/target.go
@@ -65,7 +65,7 @@ type TargetSpec struct {
 }
 
 // overrideProperties overrides all properties that are set in child into itself using reflection.
-func (spec *TargetSpec) overrideProperties(child *TargetSpec) {
+func (spec *TargetSpec) overrideProperties(child *TargetSpec) error {
 	specType := reflect.TypeOf(spec).Elem()
 	specValue := reflect.ValueOf(spec).Elem()
 	childValue := reflect.ValueOf(child).Elem()
@@ -95,14 +95,15 @@ func (spec *TargetSpec) overrideProperties(child *TargetSpec) {
 				for j := i + 1; j < dst.Len(); j++ {
 					w := dst.Index(j).String()
 					if v == w {
-						panic("duplicate value '" + v + "' in field : " + field.Name)
+						return fmt.Errorf("duplicate value '" + v + "' in field : " + field.Name)
 					}
 				}
 			}
 		default:
-			panic("unknown field type : " + kind.String())
+			return fmt.Errorf("unknown field type : " + kind.String())
 		}
 	}
+	return nil
 }
 
 // load reads a target specification from the JSON in the given io.Reader. It
@@ -117,10 +118,10 @@ func (spec *TargetSpec) load(r io.Reader) error {
 }
 
 // loadFromGivenStr loads the TargetSpec from the given string that could be:
-// - targets/ directory inside the compiler sources
-// - a relative or absolute path to custom (project specific) target specification .json file;
-//   the Inherits[] could contain the files from target folder (ex. stm32f4disco)
-//   as well as path to custom files (ex. myAwesomeProject.json)
+//   - targets/ directory inside the compiler sources
+//   - a relative or absolute path to custom (project specific) target specification .json file;
+//     the Inherits[] could contain the files from target folder (ex. stm32f4disco)
+//     as well as path to custom files (ex. myAwesomeProject.json)
 func (spec *TargetSpec) loadFromGivenStr(str string) error {
 	path := ""
 	if strings.HasSuffix(str, ".json") {
@@ -150,11 +151,17 @@ func (spec *TargetSpec) resolveInherits() error {
 		if err != nil {
 			return err
 		}
-		newSpec.overrideProperties(subtarget)
+		err = newSpec.overrideProperties(subtarget)
+		if err != nil {
+			return err
+		}
 	}
 
 	// When all properties are loaded, make sure they are properly inherited.
-	newSpec.overrideProperties(spec)
+	err := newSpec.overrideProperties(spec)
+	if err != nil {
+		return err
+	}
 	*spec = *newSpec
 
 	return nil
@@ -221,7 +228,7 @@ func LoadTarget(options *Options) (*TargetSpec, error) {
 	// it includes all parents as specified in the "inherits" key.
 	err = spec.resolveInherits()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%s : %w", options.Target, err)
 	}
 
 	if spec.Scheduler == "asyncify" {

